plugins/package-managers/node/src/main/kotlin/PackageJson.kt

```diff
@@ -74,6 +74,13 @@ private fun transformPackageJson(element: JsonElement): JsonElement {
     val entries = obj.entries.associateTo(mutableMapOf()) { it.toPair() }.apply {
         remove("license")
         put("licenses", JsonArray(licenses.map { JsonPrimitive(it) }))
+
+        if ("url" !in (obj["repository"] as? JsonObject)?.keys.orEmpty()) {
```

Comment:
This condition would remove the element also if other fields than `url` *are* present. So how about using `(obj["repository"] as? JsonObject)?.isEmpty() == true` instead which emphasizes that the whole object is empty?

Reply:
> This condition would remove the element also if other fields than `url` _are_ present.

The reason I decided for this is to also fix crashing in the same way in this case. As written in the
commit message, I believe repository nodes without `url` do not make sense (also looking at the specs) and should be discarded.

> So how about using (obj["repository"] as? JsonObject)?.isEmpty() == true instead which emphasizes that the whole object is empty?

This has a different effect. Would you prefer to keep on crashing in the above case? 

What do you think?

Reply:
> Would you prefer to keep on crashing in the above case?

Ideally we would not really crash, but create an issue about it, as I believe there's value in informing the user about broken definition files. However, as it's not easy to create an ORT issue from here without stopping deserialization, let's keep our existing code.

Reply:
> I believe there's value in informing the user about broken definition files.

I've created the more general https://github.com/oss-review-toolkit/ort/issues/9383 for this.